include/atomic_queue/atomic_queue.h

```diff
@@ -495,13 +496,17 @@ class AtomicQueueB : public AtomicQueueCommon<AtomicQueueB<T, A, NIL, MAXIMIZE_T
     friend void swap(AtomicQueueB& a, AtomicQueueB& b) noexcept {
         a.swap(b);
     }
+
+    allocator_type get_allocator() const noexcept {
+        return static_cast<allocator_type>(static_cast<const AllocatorElements&>(*this));
+    }
 };
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 template<class T, class A = std::allocator<T>, bool MAXIMIZE_THROUGHPUT = true, bool TOTAL_ORDER = false, bool SPSC = false>
 class AtomicQueueB2 : public AtomicQueueCommon<AtomicQueueB2<T, A, MAXIMIZE_THROUGHPUT, TOTAL_ORDER, SPSC>>,
-                      private A,
+                      private A, // TODO: Do we want to inherit from two allocators if we are dealing with a stateful allocator?
```

Comment:
Here it gets a bit dicey with 2 allocator types used. 

Which one should `get_allocator` return?

May be a better design is to always derive the user provided allocator type as is without rebinding and that is what `get_allocator` should return. And use the rebound copy of the allocator for allocation, in both B and B2 versions.

Would that be a valid solution?